app/Nova/User.php

```diff
@@ -809,4 +815,31 @@ private static function adminCanRun(NovaRequest $request): bool
     {
         return $request->user()->hasRole('admin');
     }
+
+    //Helper method: Graduation semester is a 6-digit code by default.
+    //Prepares semester data for display
+    private static function parseGradSemester(?string $sem): string
+    {
+        if ($sem === null) {
+            return '';
+        }
+        if (preg_match('/^[0-9]{4}0[258]$/', $sem) === false) {
+            throw new InvalidArgumentException('Invalid date code for field \'graduation_semester\'.');
```

Comment:
Return `null` or empty string here. This field is not critical; if it's not valid it's better to just leave it blank rather than throwing an error.